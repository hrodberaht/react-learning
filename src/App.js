import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ConnectedCardsDashboard from './components/CardsDashboard/CardsDashboard';
import Header from './components/Header/Header';
import Home from './components/Home/Home';

import { fetchCards } from './actions';
import NotFound from './components/NotFound/NotFound';
import ProductsDashboard from './components/ProductsDashboard/ProductsDashboard';
import Auth from './components/Auth/Auth';


export class App extends Component {
  componentDidMount() {
    const { handleFetch } = this.props;
    handleFetch();
  }

  render() {
    return (
      <Router>
        <div className="App container">
          <Header className="Header" />
          <Switch>
            <Route path="/" exact component={Home} />
            <Route path="/cards" component={ConnectedCardsDashboard} />
            <Route path="/products" component={ProductsDashboard} />
            <Route path="/login" component={Auth} />

            <Route component={NotFound} />
          </Switch>
        </div>
      </Router>
    );
  }
}

App.propTypes = {
  handleFetch: PropTypes.func.isRequired,
};

export default connect(null, { handleFetch: fetchCards })(App);
