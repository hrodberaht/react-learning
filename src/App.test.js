import React from 'react';
import { shallow } from 'enzyme';

import { App } from './App';
import Home from './components/Home/Home';
import ConnectedCardsDashboard from './components/CardsDashboard/CardsDashboard';
import NotFound from './components/NotFound/NotFound';

describe('<App />', () => {
  let wrapper;

  const props = {
    handleFetch: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<App {...props} />);
  });
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleFetch function', () => {
    expect(props.handleFetch).toHaveBeenCalled();
  });

  it('should route / should link to Home', () => {
    expect(wrapper.find('Route[exact=true][path="/"]').props().component).toEqual(Home);
  });

  it('should route /cards should link to Cards', () => {
    expect(wrapper.find('Route[path="/cards"]').props().component).toEqual(ConnectedCardsDashboard);
  });

  it('should route /cards should link to Cards', () => {
    expect(wrapper.find('Route').last().props().component).toEqual(NotFound);
  });
});
