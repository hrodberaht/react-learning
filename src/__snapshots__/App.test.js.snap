// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    handleFetch={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
>
        <Header />
        <Switch>
                <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                />
                <Route
                        component={[Function]}
                        path="/cards"
                />
                <Route
                        component={[Function]}
                />
        </Switch>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        path="/cards"
            />
            <Route
                        component={[Function]}
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                path="/cards"
/>,
              <Route
                component={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/cards",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
>
          <Header />
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              path="/cards"
                    />
                    <Route
                              component={[Function]}
                    />
          </Switch>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header />,
            <Switch>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
              <Route
                            component={[Function]}
                            path="/cards"
              />
              <Route
                            component={[Function]}
              />
</Switch>,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
/>,
                <Route
                  component={[Function]}
                  path="/cards"
/>,
                <Route
                  component={[Function]}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/cards",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
